packages/discovery-react-components/src/utils/document/documentUtils.ts

```diff
@@ -84,7 +85,26 @@ export function getTextNodeAndOffset(node: Node, offset: number): NodeOffset {
     throw new Error(`Failed to find text node. Node: ${node.textContent}, offset: ${offset}`);
   }
 
-  const textOffset = Math.max(0, offset - runningOffset);
+  let textOffset = Math.max(0, offset - runningOffset);
+
+  // If attribute 'data-orig-text' is present then
```

Comment:
do we have a test for this?

Reply:
I took a stab, but there are some differences between browser DOM and `jsdom` (as used in tests). I'll make an issue to add some tests around this.